pkg/clustermanager/cluster_manager.go

```diff
@@ -1394,3 +1407,44 @@ func (c *ClusterManager) buildSpecForCluster(ctx context.Context, clus *types.Cl
 func (c *ClusterManager) DeletePackageResources(ctx context.Context, managementCluster *types.Cluster, clusterName string) error {
 	return c.clusterClient.DeletePackageResources(ctx, managementCluster, clusterName)
 }
+
+func (c *ClusterManager) getUpgraderImagesFromBundle(ctx context.Context, cluster *types.Cluster, cl *cluster.Spec) (*corev1.ConfigMap, error) {
+	upgraderImages := make(map[string]string)
+	for _, versionBundle := range cl.Bundles.Spec.VersionsBundles {
+		eksD := versionBundle.EksD
+		eksdVersion := fmt.Sprintf("%s-eks-%s-%s", eksD.KubeVersion, eksD.ReleaseChannel, strings.Split(eksD.Name, "-")[4])
+		if _, ok := upgraderImages[eksdVersion]; !ok {
+			upgraderImages[eksdVersion] = versionBundle.Upgrader.Upgrader.URI
+		}
+	}
+
+	upgraderConfigMap, err := c.clusterClient.GetConfigMap(ctx, cluster.KubeconfigFile, constants.UpgraderConfigMapName, constants.EksaSystemNamespace)
+	if err != nil {
+		if executables.IsKubectlNotFoundError(err) {
+			return newUpgraderConfigMap(upgraderImages), nil
+		}
+		return nil, err
+	}
+
+	for version, image := range upgraderImages {
+		if _, ok := upgraderConfigMap.Data[version]; !ok {
+			upgraderConfigMap.Data[version] = image
+		}
```

Comment:
I think for this case, when a version is already in the configmap, we should override it instead of preserving the older version
```suggestion
		upgraderConfigMap.Data[version] = image
```